src/platform/packages/shared/kbn-discover-contextual-components/src/data_types/logs/components/summary_column/summary_column.tsx

```diff
@@ -98,14 +98,16 @@ const SummaryCell = ({
   );
 };
 
-const SummaryCellPopover = (props: AllSummaryColumnProps) => {
+export const SummaryCellPopover = (props: AllSummaryColumnProps) => {
   const { row, dataView, fieldFormats, onFilter, closePopover, share, core } = props;
 
   const resourceFields = createResourceFields(row, core, share);
   const shouldRenderResource = resourceFields.length > 0;
 
   const documentOverview = getLogDocumentOverview(row, { dataView, fieldFormats });
-  const { field, value } = getMessageFieldWithFallbacks(documentOverview);
+  const { field, value, formattedValue } = getMessageFieldWithFallbacks(documentOverview, {
```

Comment:
> [!TIP]
> Since the EuiCodeBlock component might receive conditionally the children props, what about pre-computing the props early and passing them into the component to prevent overlaps?
```tsx
const messageCodeBlockProps = formattedValue 
  ? { language: 'json', dangerouslySetInnerHTML: { __html: value ?? '' } }
  : { language: 'txt', children: formattedValue }

<EuiCodeBlock
  overflowHeight={100}
  paddingSize="s"
  isCopyable
  fontSize="s"
  {...messageCodeBlockProps}
/>
```

Same could be applied to the other usage in the flyout.

Reply:
Good point, this is much cleaner and easier to read 👍 Updated here: https://github.com/elastic/kibana/pull/205666/commits/a201cb6e5351ee7225a9c4df7271eed32c8a02d9.